core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -647,4 +647,42 @@ package object config {
       .stringConf
       .toSequence
       .createWithDefault(Nil)
+
+  private[spark] val KAFKA_DELEGATION_TOKEN_ENABLED =
+    ConfigBuilder("spark.kafka.delegation.token.enabled")
+      .doc("Set to 'true' for obtaining delegation token from kafka.")
+      .booleanConf
+      .createWithDefault(false)
+
+  private[spark] val KAFKA_BOOTSTRAP_SERVERS =
+    ConfigBuilder("spark.kafka.bootstrap.servers")
```

Comment:
hmm, I'm not fond of adding spark specific configs for all of these, does kafka client not read the kafka configs to know how to get tokens?  Sorry I haven't looked at the kafka implementation yet.

Reply:
Please see SPIP for details.

Reply:
I would briefly explain here tho.

Reply:
Initial token fetching is at the very beginning in the Spark process, before parsing source/sink parameters. Passing Kafka parameters now possible on source/sink which is late from delegation token point of view.

Reply:
It exists and described in the SPIP: `spark.security.credentials.kafka.enabled`

Reply:
Given the defaults the tokens fetch would be attempted if only `spark.kafka.bootstrap.servers` is defined right ? And the spark-sql-kafka libraries needs to be in the class path as well ? 

Better mention these in the docs. And make the `spark.security.credentials.kafka.enabled` default to false if it makes sense.

Also consider prefixing all the configs with `spark.security.credentials.kafka` instead of `spark.kafka`  (like `spark.security.credentials.kafka.bootstrap.servers`) to make it explicit that these are security related settings required for fetching kafka delegation tokens.